source/mir/random/generic.d

```diff
@@ -0,0 +1,384 @@
+/**
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Sebastian Wilzbach, Ilya Yaroshenko
+
+TODO:
+- automatic method to identify inflection points?
+
+*/
+module mir.random.generic;
+
+alias f_t = double;
+
+import std.stdio;
+
+import mir.random.intern;
+
+/**
+
+
+References:
+    Transformed Density Rejection with Inflection Points)
+
+    Botts, Carsten, Wolfgang Hörmann, and Josef Leydold.
+    "$(LINK2 http://epub.wu-wien.ac.at/3158/1/techreport-110.pdf,
+    Transformed density rejection with inflection points.)"
+    Statistics and Computing 23.2 (2013): 251-260.
+*/
+
+/**
+
+Split the domain of f into three types:
+
+These can be easily found by plotting `f''`.
+$(B Inflection point) can be identified by observing at which points `f''` is 0.
+
+$(UL
+
+    $(LI $(F_TILDE) is entirely concave (`f'' is entirely negative))
+    $(LI $(F_TILDE) is entirely convex (`f'' is entirely positive))
+    $(LI $(F_TILDE) contains one inflection point (`f'' intersects the x-axis once)
+)
+
+Macros:
+    F_TILDE=g
+*/
+
+auto tinflex(T...)(T bs)
```

Comment:
I would recommend that you try to pick semantically meaningful (i.e. self-documenting) names for your parameters as a matter of habit.

In cases where you're trying to match notation in the Tinflex paper, give consideration to when that's really important; if you feel the match is important, document the meaning clearly.